### PR TITLE
Add launch command with arn or roleName option

### PR DESCRIPTION
**Pull request by koudaiii**
## WHY

Add launch command with ARN or Role Name option.

ref. https://github.com/aws/aws-sdk-go/blob/master/service/ec2/examples_test.go#L6985-L6988

### CODE REVIEW

**Comment by dtan4** [comment]
Is this really pass to all cases?
- specify neither ARN nor IAM Role Name
- specify ARN only
- specify IAM Role Name only
- specify both ARN and IAM Role Name

**Comment by koudaiii** [comment]
- [x] specify neither ARN nor IAM Role Name

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \

===> Successfully launched! Instance ID: i-
```

ok,create instance.
- [x] specify ARN only

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \
       --arn $ARN \
===> Successfully launched! Instance ID: i-
```

ok, create instance.
- [x] specify IAM Role Name only

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \
       --roleName $ROLE_NAME \
===> Successfully launched! Instance ID: i-
```

ok. create instance.
- [x] specify both ARN and IAM Role Name => Not Supported

```
ec2c launch \
       --ami $AMI_ID \
       --publicip \
       --az $AVAILABILITY_ZONE \
       --type $INSTANCE_TYPE \
       --key $SSH_KEY_NAME \
       --sg $SECURITY_GROUPS \
       --subnet $SUBNET_ID \
       --userData ./user-data \
       --volumeSize $VOLUME_SIZE \
       --roleName $ROLE_NAME \
       --arn $ARN \
```

```
panic: InvalidParameterCombination: The parameter 'iamInstanceProfile.name' may not be used in combination with 'iamInstanceProfile.arn'
    status code: 400, request id: fdd5a2c1-8498-439a-a4d9-78927e04b813
```

I added validate check.

```
InvalidParameterCombination: The parameter 'roleName' and 'arn' can not be used in combination
```

plz review 🏹

File: command/launch.go [hunk 1]
```diff
@@ -64,6 +69,11 @@ func (c *LaunchCommand) Run(args []string) int {
 		flags.Parse(flags.Args()[1:])
 	}
 
+	if arn != "" && roleName != "" {
+		fmt.Fprintln(os.Stderr, "InvalidParameterCombination: The parameter 'roleName' and 'arn' can not be used in combination")
```
Comment (dtan4) [inline command/launch.go]: Is this message copied from original error of aws-sdk-go?
If so, please show it as it is.

``` go
fmt.Fprintln(os.Stderr, err)
return 1
```

(Of course, I have to treat all other errors as well as this...)

**Review by dtan4 — CHANGES_REQUESTED** [review]

File: command/launch.go (hunk 1 above)
Comment (koudaiii) [reply command/launch.go]: I see :) I will try it :) Thank you!!

Comment (koudaiii) [reply command/launch.go]: https://github.com/dtan4/ec2c/pull/10/files#diff-0380a26c6fb593cb2283592d0ba604d4R127

Already display error message. 👀 

Change "panic" to "return 1" will be next PR.

File: command/launch.go [hunk 2]
```diff
@@ -54,6 +56,8 @@ func (c *LaunchCommand) Run(args []string) int {
 	flags.StringVar(&userData, "userData", "", "User data")
 	flags.Int64Var(&volumeSize, "volumeSize", 8, "Volume size (default: 8)")
 	flags.StringVar(&volumeType, "volumeType", "gp2", "Volume type (default: gp2)")
+	flags.StringVar(&arn, "arn", "", "ARN")
```
Comment (dtan4) [inline command/launch.go]: Please explain in detail what ARN this is...

**Review by dtan4 — CHANGES_REQUESTED** [review]

File: command/launch.go (hunk 2 above)
Comment (koudaiii) [reply command/launch.go]: Amazon Resource Name (ARN) !!

Comment (dtan4) [reply command/launch.go]: Sorry for my insufficient explanation...
I'd like to ask you: what RESOURCE is this ARN attached to?

ex. IAM Policy ARN, IAM Instance Profile ARN

Comment (koudaiii) [reply command/launch.go]: Ooops! ok, I understand!!

https://github.com/aws/aws-sdk-go/blob/master/service/ec2/examples_test.go#L6985-L6987

```
        IamInstanceProfile: &ec2.IamInstanceProfileSpecification{
            Arn:  aws.String("String"),
            Name: aws.String("String"),
        },
```

`IAM Instance Profile ARN`

**Review by dtan4 — APPROVED** [review]

**Comment by dtan4** [comment]
LGTM 👍 Thanks ❗️

**Pull request merged**